modules/overrides/user_configurable_overrides.go

```diff
@@ -184,13 +184,61 @@ func (o *userConfigurableOverridesManager) setTenantLimit(userID string, limits
 }
 
 func (o *userConfigurableOverridesManager) Forwarders(userID string) []string {
-	tenantLimits, ok := o.getTenantLimits(userID)
-	if ok && tenantLimits.Forwarders != nil {
-		return *tenantLimits.Forwarders
+	if forwarders, ok := o.getTenantLimits(userID).GetForwarders(); ok {
+		return forwarders
 	}
 	return o.Interface.Forwarders(userID)
 }
 
+func (o *userConfigurableOverridesManager) MetricsGeneratorProcessors(userID string) map[string]struct{} {
+	if processors, ok := o.getTenantLimits(userID).GetMetricsGenerator().GetProcessors(); ok {
+		return processors.GetMap()
+	}
+	return o.Interface.MetricsGeneratorProcessors(userID)
+}
+
+func (o *userConfigurableOverridesManager) MetricsGeneratorDisableCollection(userID string) bool {
+	if disableCollection, ok := o.getTenantLimits(userID).GetMetricsGenerator().GetDisableCollection(); ok {
+		return disableCollection
+	}
+	return o.Interface.MetricsGeneratorDisableCollection(userID)
+}
+
+func (o *userConfigurableOverridesManager) MetricsGeneratorProcessorServiceGraphsDimensions(userID string) []string {
+	if dimensions, ok := o.getTenantLimits(userID).GetMetricsGenerator().GetProcessor().GetServiceGraphs().GetDimensions(); ok {
+		return dimensions
+	}
+	return o.Interface.MetricsGeneratorProcessorServiceGraphsDimensions(userID)
+}
+
+func (o *userConfigurableOverridesManager) MetricsGeneratorProcessorServiceGraphsEnableClientServerPrefix(userID string) bool {
+	if enableClientServerPrefix, ok := o.getTenantLimits(userID).GetMetricsGenerator().GetProcessor().GetServiceGraphs().GetEnableClientServerPrefix(); ok {
+		return enableClientServerPrefix
+	}
+	return o.Interface.MetricsGeneratorProcessorServiceGraphsEnableClientServerPrefix(userID)
+}
+
+func (o *userConfigurableOverridesManager) MetricsGeneratorProcessorServiceGraphsPeerAttributes(userID string) []string {
+	if peerAttribtues, ok := o.getTenantLimits(userID).GetMetricsGenerator().GetProcessor().GetServiceGraphs().GetPeerAttributes(); ok {
+		return peerAttribtues
+	}
+	return o.Interface.MetricsGeneratorProcessorServiceGraphsPeerAttributes(userID)
+}
+
+func (o *userConfigurableOverridesManager) MetricsGeneratorProcessorSpanMetricsDimensions(userID string) []string {
+	if dimensions, ok := o.getTenantLimits(userID).GetMetricsGenerator().GetProcessor().GetSpanMetrics().GetDimensions(); ok {
+		return dimensions
+	}
+	return o.Interface.MetricsGeneratorProcessorSpanMetricsDimensions(userID)
+}
+
+func (o *userConfigurableOverridesManager) MetricsGeneratorProcessorSpanMetricsEnableTargetInfo(userID string) bool {
+	if enableTargetInfo, ok := o.getTenantLimits(userID).GetMetricsGenerator().GetProcessor().GetSpanMetrics().GetEnableTargetInfo(); ok {
```

Comment:
It seems things in this call-chain can be nil, like `GetMetricsGenerator()`.  Is that possible?

Reply:
Yes, since the overrides can be not set I've made all of the fields pointers. So if you are fetching a nested field every step until that field could be potentially nil.

To make it easier to use I added these helper Get-functions that can deal with a nil receiver (similar to what protobuf generates for you). It's a lot of annoying boilerplate but the nested if-cases would be even worse imo.